odoo/http.py

```diff
@@ -1144,17 +1152,46 @@ def make_response(self, data, headers=None, cookies=None):
         complete response object, or the returned data will not be correctly
         interpreted by the clients.
 
-        :param basestring data: response body
+        :param str data: response body
+        :param int status: http status code
         :param headers: HTTP headers to set on the response
         :type headers: ``[(name, value)]``
         :param collections.Mapping cookies: cookies to set on the client
+        :returns: a response object.
+        :rtype: :class:`~odoo.http.Response`
         """
-        response = Response(data, headers=headers)
+        response = Response(data, status=status, headers=headers)
         if cookies:
             for k, v in cookies.items():
                 response.set_cookie(k, v)
         return response
 
+    def make_json_response(self, data, status=200, headers=None, cookies=None):
+        """ Helper for JSON responses, it json-serializes ``data`` and
+        sets the Content-Type header accordingly.
+
+        :param data: the data that will be serialized into the response body
+        :type data: anything that is json-serializable
+        :param int status: http status code
+        :param headers: HTTP headers to set on the response
+        :type headers: ``[(name, value)]``
+        :param collections.Mapping cookies: cookies to set on the client
+        :returns: a response object.
+        :rtype: :class:`~odoo.http.Response`
+            """
+        if headers is None:
+            headers = []
+
+        data = json.dumps(data, ensure_ascii=False)
+        for header_no, (header_name, _) in enumerate(headers):
+            if header_name.casefold() == 'content-type':
+                headers[header_no] = ('Content-Type', 'application/json; charset=utf-8')
```

Comment:
I don't think that's correct, if the user requests a specific content-type for the response we should let them e.g. maybe they need to provide a specific suffix or parameter, or even a content type which is only incidentally JSON e.g. `application/manifest+json`. This should only add a content-type if none is found.